ioc_copier/ioc_copier.py

```diff
@@ -0,0 +1,240 @@
+def copy_folder(file_format, ioc_name):
+    """
+    Function to handle copying folder before replacing text and names.
+    Parameters:
+        file_format - The format to use for the folder name, either ending in app
+                      or starting with ioc. in the form of an fstring.
+        ioc_name - name of the ioc.
+    Returns:
+        The path of the new folder.
+    """
+    start_path = file_format.format(f"{ioc_name}-{padded_start_copy}")
+    path = os.path.join(os.getcwd(), file_format.format(f"{ioc_name}-{padded_current_copy}"))
+    try:
+        copytree(os.path.join(os.getcwd(), start_path), os.path.join(path))
+    except FileExistsError:
+        raise FileExistsError(f"Copy {padded_current_copy} already exists, please ensure that the initial copy "
+                              f"argument is greater than the highest number IOC.") from None
+    return path
+
+
+def get_file_text(file, ioc, root):
+    """
+    function to get the text to write to a file.
+    :param file: The file to get the initial text from.
+    :param ioc: The Ioc name.
+    :param root: The root folder.
+    :return:
+        The text from the file, with the IOC number updated to the current copy.
+    """
+    with open(os.path.join(root, file), "r") as file_pointer:
+        text = file_pointer.readlines()
+    replace_text(text, ioc)
+    return text
+
+
+def write_file_text(file, root, text):
+    """
+    function to write to a file.
+    :param file: The file to write to.
+    :param root: The root folder.
+    :param text: The text to write to the file.
+    """
+    with open(os.path.join(root, file), "w") as file_pointer:
+        file_pointer.seek(0)
+        file_pointer.writelines(text)
+        file_pointer.truncate()
+
+
+def file_walk(files, ioc, root):
+    """
+    Function to walk through each file retrieved by os.walk and call necessary functions.
+    :param current_copy: The current ioc number.
+    :param start_copy: The ioc number to copy
+    :param padded_current_copy: The current ioc number with zero padding.
+    :param padded_start_copy: The ioc number to copy with zero padding.
+    :param files: The list of files to walk through.
+    :param ioc: The ioc name.
+    :param root: The root folder.
+    :return:
+    """
+    for file in files:
+        if "exe" not in file and "obj" not in file and "pdb" not in file:
+            text = get_file_text(file, ioc, root)
+            write_file_text(file, root, text)
+            rename_files(root, file, ioc)
+
+
+def folder_walk(ioc, root, sub_folder):
+    """
+    Function to walk through folders and rename them.
+    :param ioc: The ioc name.
+    :param padded_current_copy:  the current iteration of the ioc number, zero padded
+    :param padded_start_copy: the ioc number to copy, zero padded.
+    :param root:
+    :param sub_folder:
+    :return:
+    """
+    for folder in sub_folder:
+        rename_files(root, folder, ioc, padded_start_copy, padded_current_copy)
```

Comment:
There are too many positional arguments here for the `rename_files()` function call.